lib/matplotlib/text.py

```diff
@@ -89,6 +90,21 @@ def _get_textbox(text, renderer):
     return x_box, y_box, w_box, h_box
 
 
+def _get_text_metrics_with_cache(renderer, text, fontprop, ismath, dpi):
+    """Call ``renderer.get_text_width_height_descent``, caching the results."""
+    # Cached based on a copy of fontprop so that later in-place mutations of
+    # the passed-in argument do not mess up the cache. dpi is not used, but
+    # participates in cache invalidation (via the renderer).
```

Comment:
```suggestion
    # the passed-in argument do not mess up the cache.
```
Move that note down, because `dpi` is only passed on here. Clever tools will show unused in the function below and people might be tempted to remove it without consulting the context of the call site.